src/Cake.Core/IO/Arguments/SwitchArgument.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Globalization;
+
+namespace Cake.Core.IO.Arguments
+{
+    /// <summary>
+    /// Represents a argument preceded by a switch.
+    /// </summary>
+    public class SwitchArgument : IProcessArgument
+    {
+        private readonly string _switch;
+        private readonly IProcessArgument _argument;
+        private readonly string _seperator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SwitchArgument"/> class.
+        /// </summary>
+        /// <param name="switch">The switch.</param>
+        /// <param name="argument">The argument.</param>
+        /// <param name="seperator">The separator between the <paramref name="switch"/> and the <paramref name="argument"/>.</param>
+        public SwitchArgument(string @switch, IProcessArgument argument, string seperator = " ")
+        {
+            _switch = @switch;
+            _argument = argument;
+            _seperator = seperator;
+        }
+
+        /// <summary>
+        /// Render the arguments as a <see cref="System.String" />.
+        /// Sensitive information will be included.
+        /// </summary>
+        /// <returns>
+        /// A string representation of the argument.
+        /// </returns>
+        public string Render()
+        {
+            return string.Format(CultureInfo.InvariantCulture, "{0}{1}{2}", _switch, _seperator, _argument.Render());
```

Comment:
This doesn't bother me either way :+1: @RichiCoder1 if you have no objections, I will make the suggested change.